ggshield/cmd/sca/scan.py

```diff
@@ -0,0 +1,166 @@
+@click.argument(
+    "directory",
+    type=click.Path(exists=True, readable=True, path_type=Path, file_okay=False),
+    required=False,
+)
+@click.option(
+    "--ref",
+    metavar="REF",
+    help="Git reference to compare working directory to.",
+)
+@click.option(
+    "--staged",
+    is_flag=True,
+    help="Compare staged state instead of working state.",
+)
+@click.pass_context
+def scan_diff_cmd(
+    ctx: click.Context,
+    directory: Optional[Path],
+    ref: str,
+    **kwargs: Any,
+) -> int:
+    """
+    Find SCA vulnerabilities in a git working directory, compared to git history.
+    """
+    return 0
+
+
+@scan_group.command(name="all")
+@click.argument(
+    "directory",
+    type=click.Path(exists=True, readable=True, path_type=Path, file_okay=False),
+    required=False,
+)
+@click.pass_context
+def scan_all_cmd(
+    ctx: click.Context,
+    directory: Optional[Path],
+    **kwargs: Any,
+) -> Optional[DummyReturnModel]:
+    """
+    Scan a directory for SCA vulnerabilities.
+    """
+    if directory is None:
+        directory = Path().resolve()
+
+    # Adds client and required parameters to the context
+    update_context(ctx)
+
+    result = sca_scan_all(ctx, directory)
+    return result
+
+
+def update_context(ctx: click.Context) -> None:
+    """
+    Basic implementation, should be populated with required parameters later.
+    """
+    config: Config = ctx.obj["config"]
+    ctx.obj["client"] = create_client_from_config(config)
+
+    # Empty for now
+    ctx.obj["exclusion_regexes"] = set()
+
+
+def sca_scan_all(ctx: click.Context, directory: Path) -> Optional[DummyReturnModel]:
+    """
+    Scan an entire directory for SCA Vulnerabilities.
+
+    - List SCA related files with a first call to SCA compute files API
+    - Create a tar archive with the required files contents
+    - Launches the scan with a call to SCA public API
+    """
+    client = SCAClient(ctx.obj["client"])
+
+    sca_filepaths = get_sca_scan_all_filepaths(
+        directory=directory,
+        exclusion_regexes=ctx.obj["exclusion_regexes"],
+        verbose=ctx.obj["config"].verbose,
+        client=client,
+    )
+
+    # No file to scan
+    if sca_filepaths is None:
+        display_info("\nNo file to scan.")
+        return
+
+    # empty for now
+    scan_parameters = SCAScanParameters()
+
+    # Call to full scan API and get results
+    scan_result = client.sca_directory_scan_all(
+        directory,
+        sca_filepaths,
+        scan_parameters,
+        ScanContext(
+            command_path=ctx.command_path,
+            scan_mode=ScanMode.SCA_DIRECTORY,
+        ).get_http_headers(),
+    )
+
+    # TODO add SCAScanResult when merged
+    if not scan_result.success or not isinstance(scan_result, DummyReturnModel):
+        if scan_result.status_code == 401:
+            raise APIKeyCheckError(client.base_uri, "Invalid API key.")
+        display_error("\nError scanning.")
+        display_error(str(scan_result))
```

Comment:
Here you don't need to prepend with `\n`?

Reply:
I removed the `\n`, they came from some copy of iac logs.